python/tk_multi_publish2/dialog.py

```diff
@@ -787,17 +814,63 @@ def _create_master_summary_details(self):
             # the summary view's context widget
             context_key = list(current_contexts.keys())[0]
             self.ui.context_widget.set_context(current_contexts[context_key])
-            context_label_text = "Task and Entity Link to apply to all items:"
+            # Check for task_required and change UI styling accordingly
+            if not self._validate_task_required():
+                # Change task label color to SG_ALERT_COLOR
+                self.ui.context_widget.ui.task_label.setStyleSheet(
+                    "color: "
+                    + sgtk.platform.current_bundle().style_constants["SG_ALERT_COLOR"]
+                )
+                # Also change the text and color of the label
+                # Use SG_HIGHLIGHT_COLOR for better readability
+                context_label_text = (
+                    "<p>Task Required is <b>True</b> in your configuration. "
+                    "Please confirm all Tasks are assigned to continue.</p>"
+                )
+                self.ui.context_widget.ui.label.setStyleSheet(
+                    "color: "
+                    + sgtk.platform.current_bundle().style_constants[
+                        "SG_HIGHLIGHT_COLOR"
+                    ]
+                )
+            else:
+                context_label_text = "Task and Entity Link to apply to all items:"
+                # Ensure styling overrides are cleared
+                self.ui.context_widget.ui.label.setStyleSheet("")
+                self.ui.context_widget.ui.task_label.setStyleSheet("")
```

Comment:
Other than the `context_label_text` text this looks like a duplication of the code at line 847-873, I wonder if this could be turned into a method?